### PR TITLE
Update to Amber v0.14.3

### LINKED ISSUE #4: Build gh-pages for Amber 0.14.3
This is dependent on PR #3 and once done will allow #2 to be closed.

### PR DESCRIPTION
**Pull request by tomrake**
I added a small widget to print a "running on Amber" (Smalltalk version) at the top of the page.
I needed to update helios to avoid the Examples and Benchfib amber_core problems.
This is still in devel (with bower_components).

Should close #4 

If you want to see preview: https://tomrake.github.io/trysmalltalk/

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,36 +1,37 @@
...
+    "Amber",
+    "Smalltalk"
+  ],
+  "license": [
+    "MIT"
+  ],
+  "private": false,
+  "dependencies": {
+    "amber": "0.14.3",
+    "trysmalltalk": "0.1.1",
```
Comment (ghost) [inline bower.json]: But again, why you fixed trysmalltalk version (and amber version)?

**Comment by tomrake** [comment]
Perhaps I am misunderstanding  but here is my reasons.

This is task #4 to update the gh-page to 0.14.3 so I listed that as `"amber": "0.14.3"`. Was just suppose to use a bower command to set the version of amber, what am I missing?

"0.1.1" is the current trysmalltalk version. I guess I could make this no version which would mean the latest version. That would be probably better as in the future there would be no need for changes here after a release.

**Comment by ghost** [comment]
You did not put `^` at the beginning, which means, you fixed the version. No further version may be installed unless manually requested. What was your reason for _fixing_ those versions?

**Comment by ghost** [comment]
(if you simply used `bower install "amber#*" "trysmalltalk#*" --save` to install latest versions, the version would not be fixed. bower would insert `~` which is similar to `^` but the latter seems to be more sane and is adopted by npm, but not yet by bower)

**Comment by ghost** [comment]
(or, similar than previous, but without grawlixes: if you removed amber and trysmalltalk entry from bower.json altogether and ran `bower install amber trysmalltalk --save`, the version will be similarly unfixed)

**Comment by tomrake** [comment]
I changed bower.json to specify patch level changes to amber and trysmalltalk be equal or above the specified patch level. Or said another way I don't expect the build of this to need a review until breaking changes in the amber or trysmalltalk API.

**Comment by ghost** [comment]
Yes, this is the default (i'd be glad is bower switched to `^` though, same as npm did). I was wandering why it was different.

File: src/AmberDemo.st [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+! !
+
+!DemoHeader class methodsFor: 'as yet unclassified'!
+
+begin
+	self new appendToBody
+!
+
+begin: msg
+	self new demotext:msg;appendToBody
```
Comment (ghost) [inline src/AmberDemo.st]: I would put space after semicolon here.

**Comment by tomrake** [comment]
I added the space after the semicolon. Added class documentation and cleaned up code a per Coding Conventions.

**Comment by tomrake** [comment]
1. The AmberExampleDemo files are useless blank files which are now not included in the gh-pages branch, likely included during a `git add -A` command when I was adding  bower_components has changed.
2. Fixed singletab indent of class: and with: cascade in #renderOn: method.
3. Left spaces each side of `,`  binary methods because it makes them easier to see.

**Comment by hhzl** [comment]
You request for Amber ~0.14.3 though Helios is `^0.3.0` in `bower.json`. I suppose this is OK, but see
https://github.com/amber-smalltalk/helios/issues/50

**Comment by ghost** [comment]
Yeah, it won't hurt if there is ^0.3.4 to be on the safe side.

Hannes Hirzel wrote:

**Comment by tomrake** [comment]
I updated the devDependencies to helios'>=0.3.4' using the command `bower install helios#'>=0.3.4'`.
 I needed helios at 0.3.4 to code AmberDemo.st,

**Comment by ghost** [comment]
There can't be `"helios": "...."` line in both dependencies and devDependencies. Leave it only at one of them. At that one where it belongs. Do trysmalltalk need Helios when it is _run in production_? If yes, leave the first one, if not, leave the second one (I think it is not needed while run in production).

**Comment by tomrake** [comment]
1. AmberDemo.st: removed the excessive parens.
2. bower.json: removed helios spec from devDependencies (all bower command line work).

**Comment by ghost** [comment]
I am firmly convinced helios should have been removed from dependencies and only left in devDependencies. What is the use case where demo code itself needs helios  (not for development, but when it runs)?

**Comment by tomrake** [comment]
I was using the **bad practice** of editing bower.json by hand, and bad conclusions about versions of helios were created.

**Comment by ghost** [comment]
@hhzl LGTM. Maybe look over it yourself as well and merge.

**Comment by ghost** [comment]
@tomrake If you know what you're doing it's ok to edit, though I prefer to be on the safe side and do everything via issuing bower itself (it knows better what it needs to do, does the checks and whatnot).

**Pull request merged**